### PR TITLE
32-Refactor-Data

### LINKED ISSUE #32: Refactor data
**Acceptance Criteria:**
- refactor mock-data User, location, activity and Venue
- Rearrange Types and Iheritance.
- Create generic for Included in Queries/user.ts

### PR DESCRIPTION
**Pull request by JWMedeiros**
Added more mock seed data, added dynamic seeding to prisma-client, added dynamic typing to user queries

### CODE REVIEW

**Comment by rmc3408** [comment]
Missing to complete PR information.

image

File: src/data/mock-data.ts [hunk 1]
```diff
@@ -4,46 +4,120 @@ import { type Location } from '../types/location'
...
+  longitude: -30,
+},
+{
+  address: '4 Water Road',
+  phone: '1115433366',
+  city: 'Mississauga',
+  postalCode: 'L5H 1R1',
+  latitude: 12,
+  longitude: -39,
+},]
```
Comment (rmc3408) [inline src/data/mock-data.ts]: remove this comma line 30

File: src/data/mock-data.ts [hunk 2]
```diff
@@ -4,46 +4,120 @@ import { type Location } from '../types/location'
...
+  workingHoursOpen: 1692349200000, 
+  workingHoursClose: 1692381600000, 
+},
+{
+  name: 'Aquarius',
+  email: 'aquarius@hotmail.com',
+  description: 'Come to look at the fishies, stay for the burger.',
+  logo: 'http://logo.iplayed.com',
+  mainPhoto: 'http://mainphoto.iplayed.com',
+  category: <Category[]>['Aquatic'],
```
Comment (rmc3408) [inline src/data/mock-data.ts]: Can you create one array with 2 items. ??

[aerial,acquatic]

File: src/data/mock-data.ts [hunk 3]
```diff
@@ -4,46 +4,120 @@ import { type Location } from '../types/location'
...
+},
+{
+  name: 'SkyPalace',
+  email: 'skypalace@gmail.com',
+  description: 'Fly with the birds, feel the wind in your hair.',
+  logo: 'http://logo.iplayed.com',
+  mainPhoto: 'http://mainphoto.iplayed.com',
+  category: <Category[]>['Aerial'],
+  isAcessibility: false,
+  workingHoursOpen: 1692349200000,
```
Comment (rmc3408) [inline src/data/mock-data.ts]: can you change to a new time for future testing ? this is epoch time 

workingHoursOpen and workingHoursClose

File: src/config/prisma/prisma-client.ts [hunk 4]
```diff
@@ -15,39 +15,40 @@ export async function startORM() {
...
-              activities: {
-                create: [
-                  {
-                    ...activityData,
-                    location: {
-                      create: locationData,
-                    },
-                  },
-                ],
+      for (let i=0; i<userData.length; i++){
```
Comment (rmc3408) [inline src/config/prisma/prisma-client.ts]: space between i<userData.

**Review by rmc3408 — COMMENTED** [review]
LGTM. Just few changes some data info for us in the future and tiny formatting issues.

File: src/config/prisma/prisma-client.ts [hunk 5]
```diff
@@ -15,39 +15,40 @@ export async function startORM() {
...
-              activities: true,
+          include: {
+            venue: {
             },
           },
-        },
-      })
-
-      // await prisma.$disconnect()
-      // shutDown('database successful seeded')
```
Comment (rmc3408) [inline src/config/prisma/prisma-client.ts:50]: Add this code back and running please.

**Review by rmc3408 — COMMENTED** [review]
Please add back the code for SHUTDOWN and running (un-comment out)

File: src/config/prisma/prisma-client.ts (hunk 5 above)
Comment (JWMedeiros) [reply src/config/prisma/prisma-client.ts:50]: Its added back in. Resolving the conversation for this.

File: src/data/mock-data.ts (hunk 1 above)
Comment (JWMedeiros) [reply src/data/mock-data.ts]: I ran the linter on this file, if you take the commas out it actually ends up putting the commas back in, but it spaces the ending bracket out better.

File: src/data/mock-data.ts (hunk 2 above)
Comment (JWMedeiros) [reply src/data/mock-data.ts]: I added this and it does in fact work. I added second categories for venues 2 and 3.

File: src/config/prisma/prisma-client.ts (hunk 4 above)
Comment (JWMedeiros) [reply src/config/prisma/prisma-client.ts]: Linter cleaned that up as well.

File: src/data/mock-data.ts (hunk 3 above)
Comment (JWMedeiros) [reply src/data/mock-data.ts]: I changed the working hours for the second and third venus, and also changed activity times for activities 2 and 3 for testing.

**Review by rmc3408 — APPROVED** [review]
LGTM , ready to merge

**Pull request merged**